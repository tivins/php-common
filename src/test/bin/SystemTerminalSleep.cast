{"version":2,"width":80,"height":24,"timestamp":1674319176,"env":{"SHELL":"\/bin\/bash","TERM":"xterm-256color"}}
[0,"o","\u0007\u001b[01;32mme@hostname\u001b[00m:\u001b[01;34m\/path\/run\u001b[00m$ run.php\r\n"]
[0.020333051681518555,"o","Sleeping 2 seconds...\r\n"]
[1.0203251838684082,"o","\u001b[1A\r\u001b[KSleeping 1 seconds...\r\n"]
[2.020418167114258,"o","\u001b[1A\r\u001b[KSleeping 0 seconds...\r\n"]
[3.0205471515655518,"o","\u001b[1A\r\u001b[KDone.\r\n"]