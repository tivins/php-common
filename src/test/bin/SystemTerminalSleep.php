<?php
use Tivins\Core\System\Terminal;
require 'vendor/autoload.php';
Terminal::sleep(2);
Terminal::sleep(2, "Remains: %.2f seconds.");
echo "Done.", PHP_EOL;
