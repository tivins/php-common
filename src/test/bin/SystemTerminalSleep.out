Sleeping 2.0 seconds...
[1A[KSleeping 2.0 seconds...
[1A[KSleeping 2.0 seconds...
[1A[KSleeping 2.0 seconds...
[1A[KSleeping 2.0 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.9 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.8 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.7 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.6 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.5 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.4 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.3 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.2 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.1 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 1.0 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.9 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.8 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.7 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.6 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.5 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.4 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.3 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.2 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.1 seconds...
[1A[KSleeping 0.0 seconds...
[1A[KSleeping 0.0 seconds...
[1A[KSleeping 0.0 seconds...
[1A[KSleeping 0.0 seconds...
[1A[KSleeping 0.0 seconds...
[1A[KRemains: 2.00 seconds.
[1A[KRemains: 1.99 seconds.
[1A[KRemains: 1.98 seconds.
[1A[KRemains: 1.97 seconds.
[1A[KRemains: 1.96 seconds.
[1A[KRemains: 1.95 seconds.
[1A[KRemains: 1.94 seconds.
[1A[KRemains: 1.93 seconds.
[1A[KRemains: 1.92 seconds.
[1A[KRemains: 1.91 seconds.
[1A[KRemains: 1.90 seconds.
[1A[KRemains: 1.89 seconds.
[1A[KRemains: 1.88 seconds.
[1A[KRemains: 1.87 seconds.
[1A[KRemains: 1.86 seconds.
[1A[KRemains: 1.85 seconds.
[1A[KRemains: 1.84 seconds.
[1A[KRemains: 1.83 seconds.
[1A[KRemains: 1.82 seconds.
[1A[KRemains: 1.81 seconds.
[1A[KRemains: 1.80 seconds.
[1A[KRemains: 1.79 seconds.
[1A[KRemains: 1.78 seconds.
[1A[KRemains: 1.77 seconds.
[1A[KRemains: 1.76 seconds.
[1A[KRemains: 1.75 seconds.
[1A[KRemains: 1.74 seconds.
[1A[KRemains: 1.73 seconds.
[1A[KRemains: 1.72 seconds.
[1A[KRemains: 1.71 seconds.
[1A[KRemains: 1.70 seconds.
[1A[KRemains: 1.69 seconds.
[1A[KRemains: 1.68 seconds.
[1A[KRemains: 1.67 seconds.
[1A[KRemains: 1.66 seconds.
[1A[KRemains: 1.65 seconds.
[1A[KRemains: 1.64 seconds.
[1A[KRemains: 1.63 seconds.
[1A[KRemains: 1.62 seconds.
[1A[KRemains: 1.61 seconds.
[1A[KRemains: 1.60 seconds.
[1A[KRemains: 1.59 seconds.
[1A[KRemains: 1.58 seconds.
[1A[KRemains: 1.57 seconds.
[1A[KRemains: 1.56 seconds.
[1A[KRemains: 1.55 seconds.
[1A[KRemains: 1.54 seconds.
[1A[KRemains: 1.53 seconds.
[1A[KRemains: 1.52 seconds.
[1A[KRemains: 1.51 seconds.
[1A[KRemains: 1.50 seconds.
[1A[KRemains: 1.49 seconds.
[1A[KRemains: 1.48 seconds.
[1A[KRemains: 1.47 seconds.
[1A[KRemains: 1.46 seconds.
[1A[KRemains: 1.45 seconds.
[1A[KRemains: 1.44 seconds.
[1A[KRemains: 1.43 seconds.
[1A[KRemains: 1.42 seconds.
[1A[KRemains: 1.41 seconds.
[1A[KRemains: 1.40 seconds.
[1A[KRemains: 1.39 seconds.
[1A[KRemains: 1.38 seconds.
[1A[KRemains: 1.37 seconds.
[1A[KRemains: 1.36 seconds.
[1A[KRemains: 1.35 seconds.
[1A[KRemains: 1.34 seconds.
[1A[KRemains: 1.33 seconds.
[1A[KRemains: 1.32 seconds.
[1A[KRemains: 1.31 seconds.
[1A[KRemains: 1.30 seconds.
[1A[KRemains: 1.29 seconds.
[1A[KRemains: 1.28 seconds.
[1A[KRemains: 1.27 seconds.
[1A[KRemains: 1.26 seconds.
[1A[KRemains: 1.25 seconds.
[1A[KRemains: 1.24 seconds.
[1A[KRemains: 1.23 seconds.
[1A[KRemains: 1.22 seconds.
[1A[KRemains: 1.21 seconds.
[1A[KRemains: 1.20 seconds.
[1A[KRemains: 1.19 seconds.
[1A[KRemains: 1.18 seconds.
[1A[KRemains: 1.17 seconds.
[1A[KRemains: 1.16 seconds.
[1A[KRemains: 1.15 seconds.
[1A[KRemains: 1.14 seconds.
[1A[KRemains: 1.13 seconds.
[1A[KRemains: 1.12 seconds.
[1A[KRemains: 1.11 seconds.
[1A[KRemains: 1.10 seconds.
[1A[KRemains: 1.09 seconds.
[1A[KRemains: 1.08 seconds.
[1A[KRemains: 1.07 seconds.
[1A[KRemains: 1.06 seconds.
[1A[KRemains: 1.05 seconds.
[1A[KRemains: 1.04 seconds.
[1A[KRemains: 1.03 seconds.
[1A[KRemains: 1.02 seconds.
[1A[KRemains: 1.01 seconds.
[1A[KRemains: 1.00 seconds.
[1A[KRemains: 0.99 seconds.
[1A[KRemains: 0.98 seconds.
[1A[KRemains: 0.97 seconds.
[1A[KRemains: 0.96 seconds.
[1A[KRemains: 0.95 seconds.
[1A[KRemains: 0.94 seconds.
[1A[KRemains: 0.93 seconds.
[1A[KRemains: 0.92 seconds.
[1A[KRemains: 0.91 seconds.
[1A[KRemains: 0.90 seconds.
[1A[KRemains: 0.89 seconds.
[1A[KRemains: 0.88 seconds.
[1A[KRemains: 0.87 seconds.
[1A[KRemains: 0.86 seconds.
[1A[KRemains: 0.85 seconds.
[1A[KRemains: 0.84 seconds.
[1A[KRemains: 0.83 seconds.
[1A[KRemains: 0.82 seconds.
[1A[KRemains: 0.81 seconds.
[1A[KRemains: 0.80 seconds.
[1A[KRemains: 0.79 seconds.
[1A[KRemains: 0.78 seconds.
[1A[KRemains: 0.77 seconds.
[1A[KRemains: 0.76 seconds.
[1A[KRemains: 0.75 seconds.
[1A[KRemains: 0.74 seconds.
[1A[KRemains: 0.73 seconds.
[1A[KRemains: 0.72 seconds.
[1A[KRemains: 0.71 seconds.
[1A[KRemains: 0.70 seconds.
[1A[KRemains: 0.69 seconds.
[1A[KRemains: 0.68 seconds.
[1A[KRemains: 0.67 seconds.
[1A[KRemains: 0.66 seconds.
[1A[KRemains: 0.65 seconds.
[1A[KRemains: 0.64 seconds.
[1A[KRemains: 0.63 seconds.
[1A[KRemains: 0.62 seconds.
[1A[KRemains: 0.61 seconds.
[1A[KRemains: 0.60 seconds.
[1A[KRemains: 0.59 seconds.
[1A[KRemains: 0.58 seconds.
[1A[KRemains: 0.57 seconds.
[1A[KRemains: 0.56 seconds.
[1A[KRemains: 0.55 seconds.
[1A[KRemains: 0.54 seconds.
[1A[KRemains: 0.53 seconds.
[1A[KRemains: 0.52 seconds.
[1A[KRemains: 0.51 seconds.
[1A[KRemains: 0.50 seconds.
[1A[KRemains: 0.49 seconds.
[1A[KRemains: 0.48 seconds.
[1A[KRemains: 0.47 seconds.
[1A[KRemains: 0.46 seconds.
[1A[KRemains: 0.45 seconds.
[1A[KRemains: 0.44 seconds.
[1A[KRemains: 0.43 seconds.
[1A[KRemains: 0.42 seconds.
[1A[KRemains: 0.41 seconds.
[1A[KRemains: 0.40 seconds.
[1A[KRemains: 0.39 seconds.
[1A[KRemains: 0.38 seconds.
[1A[KRemains: 0.37 seconds.
[1A[KRemains: 0.36 seconds.
[1A[KRemains: 0.35 seconds.
[1A[KRemains: 0.34 seconds.
[1A[KRemains: 0.33 seconds.
[1A[KRemains: 0.32 seconds.
[1A[KRemains: 0.31 seconds.
[1A[KRemains: 0.30 seconds.
[1A[KRemains: 0.29 seconds.
[1A[KRemains: 0.28 seconds.
[1A[KRemains: 0.27 seconds.
[1A[KRemains: 0.26 seconds.
[1A[KRemains: 0.25 seconds.
[1A[KRemains: 0.24 seconds.
[1A[KRemains: 0.23 seconds.
[1A[KRemains: 0.22 seconds.
[1A[KRemains: 0.21 seconds.
[1A[KRemains: 0.20 seconds.
[1A[KRemains: 0.19 seconds.
[1A[KRemains: 0.18 seconds.
[1A[KRemains: 0.17 seconds.
[1A[KRemains: 0.16 seconds.
[1A[KRemains: 0.15 seconds.
[1A[KRemains: 0.14 seconds.
[1A[KRemains: 0.13 seconds.
[1A[KRemains: 0.12 seconds.
[1A[KRemains: 0.11 seconds.
[1A[KRemains: 0.10 seconds.
[1A[KRemains: 0.09 seconds.
[1A[KRemains: 0.08 seconds.
[1A[KRemains: 0.07 seconds.
[1A[KRemains: 0.06 seconds.
[1A[KRemains: 0.05 seconds.
[1A[KRemains: 0.04 seconds.
[1A[KRemains: 0.03 seconds.
[1A[KRemains: 0.02 seconds.
[1A[KRemains: 0.01 seconds.
[1A[KDone.
